### DISCUSSION TITLE
Why the stack trace shows workflow being blocked on an activity

### DISCUSSION THREAD

**Discussion by zhangkyou** [uber/cadence#4368] [Q&A] [answered]
I got a problem in workflow executing.
I have two activities in the workflow, the first activity's output must be the second acitivty's input.
The scene is when Get the first acitivty's output, it blocked(the activity pending in state STARTED), the stack trace in cadecen-web is:
![image](https://user-images.githubusercontent.com/7250068/129508240-bf89081c-361e-4a39-97fd-2e419e72579d.png)
The consequence is that the workflow run continuous unsuccess.Always failed in the reason:"cadenceInternal:Timeout START_TO_CLOSE"

**Comment by longquanzheng** [comment, marked as answer]
Activity pending means it’s either waiting for activity worker to pick up the task(scheduled but not started) or to complete the activity (started but not completed) 

In your case it’s waiting for activity to complete. 

Also make sure you set the right retry policy and timeout for it so that it can tolerate activity worker failure/restart

**Reply by zhangkyou** [reply]
I set the activity retry policy like this, but it also blocked in activity.
![image](https://user-images.githubusercontent.com/7250068/129510212-33576152-3248-4547-8f6b-3995eb68e982.png)
workflow retry policy like this:
![image](https://user-images.githubusercontent.com/7250068/129510254-d03e7a9b-1ac6-49fb-898c-b961cfe07e53.png)

**Reply by longquanzheng** [reply]
Okay I missed the last part of the question that you said the workflow eventually failed at “start to closed timeout “. 

This is expected for workflow. The only issue is in your activity implementation. There must be something wrong there that it couldn’t complete within the timeout. You probably need to add more logging to investigate if local debug is not possible

**Reply by zhangkyou** [reply]
what casued the "blocked on chan-1.Receive" in stack trace?
why it blocked in get return value of the activity?

**Reply by longquanzheng** [reply]
Check clickhouse.go:37 
It’s blocked on that line. 

I believe it’s blocked on activity result.

“ blocked on chan-1.Receive" is internal details of the library. Cadence go client library uses channel to implement the waiting for activity results.
